django/transformer/reference_binder.py

```diff
@@ -0,0 +1,97 @@
+                logger.debug(f"reference at path {ref} has no type")
+                return
+            if not (identifier := ref.get("identifier")):
+                logger.debug(f"reference at path {ref} has no identifier")
+                return
+
+            # add the "literal reference"
+            # (https://www.hl7.org/fhir/references-definitions.html#Reference.reference)
+            try:
+                ref["reference"] = self.identifier_to_reference(identifier, reference_type)
+                logger.debug(
+                    {
+                        "message": f"reference {ref['reference']} of resource {object_id} resolved",
+                        "resource_id": resource_id,
+                    },
+                )
+            except (ValueError, KeyError) as e:
+                logger.warning(
+                    f"incomplete identifier on reference of type "
+                    f"{reference_type} at path {ref} of resource {object_id}: {e}"
+                )
+
+        def rec_bind_existing_reference(fhir_object, reference_path: List[str], sub_path=""):
+            if reference_path:
+                # We need to go down the document to find the reference.
+                # reference_path is a list of string so that at the end of each element,
+                # we have an array in the fhir document. For instance, if reference_path
+                # is ["item", "answer.valueReference"], the fhir doc will look like
+                # { item: [ { answer: { valueReference: ... } } ] }.
+                # We want to traverse all the elements of each array that could have
+                # a reference at a leaf.
+                sub_fhir_object = fhir_object[reference_path[0]]
+                sub_path = f"{sub_path}.{reference_path[0]}" if sub_path else reference_path[0]
+                if isinstance(sub_fhir_object, list):
+                    for ind, sub_fhir_el in enumerate(sub_fhir_object):
+                        rec_bind_existing_reference(sub_fhir_el, reference_path[1:], f"{sub_path}.{ind}")
+                else:
+                    rec_bind_existing_reference(sub_fhir_object, reference_path[1:], sub_path)
+            else:
+                # We have isolated the reference so we can now bind it.
+                # If it's an array, we want to perform the binding for all the
+                # elements.
+                if isinstance(fhir_object, list):
+                    for ind, sub_fhir_el in enumerate(fhir_object):
+                        bind(sub_fhir_el)
+                else:
+                    bind(fhir_object)
+
+        rec_bind_existing_reference(fhir_object, reference_path)
+
+    @staticmethod
+    def identifier_to_reference(identifier, reference_type: str) -> str:
+        if not (system := identifier.get("system")):
+            raise KeyError
+        if not (value := identifier.get("value")):
+            raise KeyError
+        namespace = UUID(system[-36:])
```

Comment:
Why `[-36:]`?

Reply:
That's the fixed length of a UUID. We can drop this as soon as we get a proper uuid input and not a url of format http://terminology.arkhn.org/.